Generics.playground/Contents.swift

```diff
@@ -1,3 +1,55 @@
 import UIKit
 
-var str = "Hello, playground"
+struct CountedSet <Element: Hashable> {
+    
+    private(set) var dictionary = [Element: Int]()
+    
+    var count: Int? {
+        if dictionary.isEmpty {
+            return nil
+        } else {
+            return dictionary.count
+        }
+    }
+    
+    mutating func insert(element: Element) {
+        dictionary[element] = (dictionary[element] ?? 0) + 1
```

Comment:
Maybe use something like this, which is a bit more readable:
```suggestion
        dictionary[element] = dictionary[element, default: 0] + 1
```